src/main/java/com/webapp/TextBook/viewModel/apiViewModel/StudentBookInfo.java

```diff
@@ -0,0 +1,164 @@
+            STUDENT_BOOK_INFO.setStudentInfo(StudentInfo.createApiFromJson(
+                    DATA.getJSONObject(
+                            StudentBookInfo.NOMINAL_STUDENT_INFO
+                    )
+            ).orElseThrow());
+
+            // sets barcode
+            STUDENT_BOOK_INFO.setBarCode(DATA.getString(StudentBookInfo.NOMINAL_BARCODE));
+        }
+        catch(JSONException ex){
+            // will never occur
+            System.out.println("Internal Error (StudentBookInfo -- StudentBookInfo's ValueStateSetter):\n" +
+                    "an error occurred that should be caught, thrown and handled internally. Please check stack trace " +
+                    "for more info." + ex.getLocalizedMessage());
+        }
+
+    });
+
+    /**
+     * Two Constuctors
+     * 1. Blank
+     * 2. Takes in both non-static inputs
+     */
+    public StudentBookInfo(){}
+
+    public StudentBookInfo(String barCode, StudentInfo studentInfo){
+        this.barCode = barCode;
+        this.studentInfo = studentInfo;
+    }
+
+    /**
+     * All relevant getters and setters
+     */
+    public String getBarCode() {
+        return barCode;
+    }
+
+    public void setBarCode(String barCode) {
+        this.barCode = barCode;
+    }
+
+    public StudentInfo getStudentInfo() {
+        return studentInfo;
+    }
+
+    public void setStudentInfo(StudentInfo studentInfo) {
+        this.studentInfo = studentInfo;
+    }
+
+    /**
+     * <p>
+     *     From the JSONObject (should be JSON composition/reflection of StudentInfo's state)
+     *     and an optional Supplier, render and sets a StudentBookInfo object to the reflective/corresponding
+     *     state within the JSONObject.
+     * </p>
+     * @param jsonObject JSONObject which is the composition of the state that defines a StudentBookInfo
+     * @param studentBookInfoSupplier Supplier(type param: StudentBookInfo) optional function to invoke to supplement
+     *                              a StudentInfo to use in creation and state setting.
+     * @return Optional(type param: StudentBookInfo) holding the ApiViewModel object. Will be empty if
+     * a) JSONObject does not hold the necessary state for a full state setting
+     * b) Supplier(type param: StudentBookInfo) throws an internal error
+     */
+    public static @NotNull Optional<StudentBookInfo> createApiFromJson(
+            @NotNull final JSONObject jsonObject,
+            @Nullable final Supplier<StudentBookInfo> studentBookInfoSupplier){
+
+        // creates a local supplier based on the predicate if the given supplier is null or not
+        // use the blank constructor as a default
+        final Supplier<StudentBookInfo> localStudentBookInfoSupplier = studentBookInfoSupplier == null ?
+                StudentBookInfo::new :
+                studentBookInfoSupplier;
+
+        // invokes parent's static helper to set the state for a LoginUserInfo
```

Comment:
student book info not login user info

Reply:
https://github.com/NicholasPierce1/TextBookServices/commit/8c720794a1054e6fa64d8cf9b922c47f3a26bdd4